Source/RxCocoa/UIScrollView+reachedBottom.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  UIScrollView+reachedBottom.swift
+//  RxSwiftExt
+//
+//  Created by Anton Nazarov on 09/05/2019.
+//  Copyright © 2019 RxSwift Community. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxCocoa
+
+public extension Reactive where Base: UIScrollView {
+    /**
+     Shows if the bottom of the UIScrollView is reached.
+     - parameter offset: A threshhold indicating the bottom of the UIScrollView.
+     - returns: ControlEvent that emits when the bottom of the base UIScrollView is reached.
+     */
+    func reachedBottom(offset: CGFloat = 0.0) -> ControlEvent<Void> {
+        let source = contentOffset.filterMap { contentOffset -> FilterMap<Void> in
```

Comment:
I think it would be better if we are not using custom operators in custom operators. 
I would like to avoid this. My suggestion would be

```swift 
func reachedBottom(offset: CGFloat = 0.0) -> ControlEvent<Void> {
        let observable = contentOffset
            .map { [weak base] contentOffset -> Bool in
                guard let scrollView = base else { return false }
                let visibleHeight = scrollView.frame.height
                    - scrollView.contentInset.top
                    - scrollView.contentInset.bottom
                let y = contentOffset.y + scrollView.contentInset.top
                let threshold = max(offset, scrollView.contentSize.height - visibleHeight)

                return y > threshold
            }
            .distinctUntilChanged()
            .skipWhile { $0 == true }
            .map { _ in }

        return ControlEvent(events: observable)
```

Reply:
The argument about atomic operators is not obvious for me. There is a library of operators, why can't we u use them in implementation on new operators?
Thank you for your implementation, but I know how to replace `filterMap`. Two lines of code. However, I can't see any pros in these code duplication.
BTW, `[weak self]` is not needed, there is no reference cycle here. I check it with debugging mode (`RxSwiftResources.Count`)

Reply:
> There are cases that I wouldn't like to import the whole library, but just grab the part that I am interested. 
You can't do it with `fromAsync` operator for instance.
If there is a requirement "Don't use other operators to implement new one" it should be discussed and placed to CONTRIBUTING.md

Reply:
I think if we're talking about two lines of code, It's better to not reuse internal operators that aren't part of core. Many people copy and paste operators from the source code since they don't want to take in a full library (another example of this is https://github.com/SwifterSwift/SwifterSwift)

Reply:
In general - I think that unless it's a necessity I'd rather not couple RxSwiftExt operators with other RxSwiftExt operators

Reply:
Where do you see the code duplication?

I gave one argument: 

> There are cases that I/others wouldn't like to import the whole library, but just grab the part that I/they care.

Reply:
> If there is a requirement "Don't use other operators to implement new one" it should be discussed and placed to CONTRIBUTING.md

I don't mind adding it to CONTRIBUTING.md. Unless it's crucial I don't see a reason to reuse these internal operators.

Reply:
Ok, let's add it there please. Could i add "You should not use operators from RxSwiftExt to implement new operators" in these PR or it's better to create another?

Reply:
I'll take care of it.

Reply:
Tnank you :)

Reply:
Replaced `.filterMap` on `.filter` + `.map`